python/common/org/python/types/Str.java

```diff
@@ -287,6 +287,9 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
         else if (other instanceof org.python.types.Float) {
             throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type '" + other.typeName() + "'");
         }
+        else if (other instanceof org.python.types.Str) {
+            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type '" + other.typeName() + "'");
+        }
```

Comment:
Let me change that, and run all the test. If they pass I believe a else statement would cut down on number of statements needed. 

From:  Russell Keith-Magee <notifications@github.com>
Reply-To:  pybee/voc <reply@reply.github.com>
Date:  Wednesday, January 6, 2016 at 7:35 PM
To:  pybee/voc <voc@noreply.github.com>
Cc:  Josh Stephens <bsdtux@gmail.com>
Subject:  Re: [voc] Implemented Str * Str (#72)


In python/common/org/python/types/Str.java:
> @@ -287,6 +287,9 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
>          else if (other instanceof org.python.types.Float) {
>              throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type '" + other.typeName() + "'");
>          }
> +        else if (other instanceof org.python.types.Str) {
> +            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type '" + other.typeName() + "'");
> +        }
Is there any reason that this can't just be else { raise TypeError }? Are there any error messages here other than "can't multiply sequence by non-int of type ..."? This might mean we end up with a bunch of other passing multiply tests as a side effect.

—
Reply to this email directly or view it on GitHub.